vault/generate_root.go

```diff
@@ -263,34 +264,67 @@ func (c *Core) GenerateRootUpdate(ctx context.Context, key []byte, nonce string,
 		}, nil
 	}
 
-	// Recover the master key
-	var masterKey []byte
+	// Combine the key parts
+	var combinedKey []byte
 	if config.SecretThreshold == 1 {
-		masterKey = c.generateRootProgress[0]
+		combinedKey = c.generateRootProgress[0]
 		c.generateRootProgress = nil
 	} else {
-		masterKey, err = shamir.Combine(c.generateRootProgress)
+		combinedKey, err = shamir.Combine(c.generateRootProgress)
 		c.generateRootProgress = nil
 		if err != nil {
 			return nil, errwrap.Wrapf("failed to compute master key: {{err}}", err)
 		}
 	}
 
 	switch {
```

Comment:
I feel like it would be less disruptive/cleaner to restore the original code here: first we do the verify recovery or master key as before.  Then we check for recovery mode and if enabled unseal, either with the master key we were provided or using the stored keys.  Then we go on to generate the requested token, as before.

Reply:
Ah, true enough.  This will be an ugly merge with my shamir-as-kek changes though.

Reply:
Sorry for that. Hope that it is not bad as you envision.

Reply:
Not your fault.  And I'm sure it won't be too bad.  Just fretful because the PR is already a month old and I already know of some difficult conflicts I'll be facing.